build/templates/session.py.mako

```diff
@@ -297,15 +235,17 @@ class Session(_SessionBase):
         ${p['python_name']} = _converters.${p['python_api_converter_name']}(${p['python_name']})
 %   endif
 % endfor
-        self._library = _library_singleton.get()
-        self._encoding = 'windows-1251'
+        self._library_interpreter = _library_interpreter.LibraryInterpreter(encoding='windows-1251')
```

Comment:
I think the interpreter should own the native session handle. In the gRPC case the equivalent interpreter will own the gRPC session handle. I don't like how the session writes directly to it, though.

Reply:
>In the gRPC case the equivalent interpreter will own the gRPC session handle.

Why? Can we change that?

Reply:
I tried it out. I also moved `encoding` back to Session, or else we can't possibly cache the LibraryInterpreter because nitclk allows the user to set the encoding. (But more on caching in another comment...) I'm not a fan of passing the session handle (and encoding) around everywhere, but let me know what you think. I can revert this change easily enough.

Reply:
I would argue that passing the session/vi and encoding to each call seems anti-object-oriented. I don't see the value in the singleton interpreter. @ni-jfitzger @marcoskirsch please provide rationale on the desire for this.

Reply:
> I also moved encoding back to Session, or else we can't possibly cache the LibraryInterpreter because nitclk allows the user to set the encoding.

Are we caching it? Do we want to cache it?
Why, oh why was that argument added to SessionReference when we don't use it anywhere?

Reply:
> I would argue that passing the session/vi and encoding to each call seems anti-object-oriented. 

It depends on how you define the class' scope. Library class represenets a DLL. Session represents a driver session. LibraryInterpreter represents... a DLL but interpreted for Python!

This is not the case today, but imagine if our libraries had global functions such as "initialize library" and "finalize library" (we've discussed these for years). Where would that go?

In any case... given that we want to have the ability to sway LibraryInterpreted for one that wraps the gRPC client code, I agree with you that we are better off keeping the session handle as a member of the LibraryInterpreter.

Reply:
> I agree with you that we are better off keeping the session handle as a member of the LibraryInterpreter.

Okay, I'll revert that commit that moved them to Session.

Reply:
Reverted the move-stuff-to-Session commit.